src/main/java/com/box/sdk/Metadata.java

```diff
@@ -0,0 +1,163 @@
+    public String getID() {
+        return this.values.get("$id").asString();
+    }
+
+    /**
+     * Returns the metadata type.
+     * @return the metadata type.
+     */
+    public String getTypeName() {
+        return this.values.get("$type").asString();
+    }
+
+    /**
+     * Returns the parent object ID (typically the file ID).
+     * @return the parent object ID.
+     */
+    public String getParentID() {
+        return this.values.get("$parent").asString();
+    }
+
+    /**
+     * Adds a new metdata value.
+     * @param path The path that designates the key. Must be prefixed with a "/".
+     * @param value The value.
+     * @return this metadata object.
+     */
+    public Metadata add(String path, String value) {
+        this.values.add(this.pathToProperty(path), value);
+        this.addOp("add", path, value);
+        return this;
+    }
+
+    /**
+     * Replaces an existing metdata value.
+     * @param path The path that designates the key. Must be prefixed with a "/".
+     * @param value The value.
+     * @return this metadata object.
+     */
+    public Metadata replace(String path, String value) {
+        this.values.set(this.pathToProperty(path), value);
+        this.addOp("replace", path, value);
+        return this;
+    }
+
+    /**
+     * Removes an existing metadata value.
+     * @param path The path that designates the key. Must be prefixed with a "/".
+     * @return this metadata object.
+     */
+    public Metadata remove(String path) {
+        this.values.remove(this.pathToProperty(path));
+        this.addOp("remove", path, null);
+        return this;
+    }
+
+    /**
+     * Tests that a property has the expected value.
+     * @param path The path that designates the key. Must be prefixed with a "/".
+     * @param value The expected value.
+     * @return this metadata object.
+     */
+    public Metadata test(String path, String value) {
+        this.addOp("test", path, value);
+        return this;
+    }
+
+    /**
+     * Returns a value.
+     * @param key The metadata property name.
+     * @return The metadata property value.
+     */
+    public String get(String key) {
+        return this.values.get(key).asString();
+    }
+
+    /**
+     * Returns the JSON patch string with all operations.
+     * @return the JSON patch string.
+     */
+    public String getPatch() {
+        if (this.operations == null) {
+            return "[]";
+        }
+        return this.operations.toString();
+    }
+
+    /**
+     * Returns the JSON representation of this metadata.
+     * @return the JSON representation of this metadata.
+     */
+    @Override
+    public String toString() {
+        return this.values.toString();
+    }
+
+    /**
+     * Converts a JSON patch path to a JSON property name.
+     * @param path The path that designates the key.  Must be prefixed with a "/".
+     * @return The JSON property name.
+     */
+    private String pathToProperty(String path) {
```

Comment:
Does the properties type support nested JSON objects? This method is okay for now if metadata can only be a string:string key-value.